compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -691,6 +691,34 @@ fn print_crate_info(
                 };
                 println_info!("{}", passes::get_crate_name(sess, attrs));
             }
+            CrateRootLintLevels => {
+                let Some(attrs) = attrs.as_ref() else {
+                    // no crate attributes, print out an error and exit
+                    return Compilation::Continue;
+                };
```

Comment:
Question [NO-ATTR 1/2]: wait so, if there are no crate-level attributes, this would print an error and not just return empty string?

Reply:
No crate-attributes means that we weren't able to parse them, not that we don't support a source file without crate attributes, we do support that just fine, `attrs` would just be a empty vec.

Reply:
Ah of course, right.